File: Q6.vb
Module Module1

    ' Base class
    Class Animal
        Public Sub Eat()
            Console.WriteLine("The animal is eating.")
        End Sub
    End Class

    ' Single inheritance: Dog inherits from Animal
    Class Dog
        Inherits Animal

        Public Sub Bark()
            Console.WriteLine("The dog is barking.")
        End Sub
    End Class

    ' Multi-level inheritance: Bulldog inherits from Dog
    Class Bulldog
        Inherits Dog

        Public Sub ShowBreed()
            Console.WriteLine("This is a Bulldog.")
        End Sub
    End Class

    Sub Main()
        ' Single Inheritance
        Dim myDog As New Dog()
        myDog.Eat() ' From Animal class
        myDog.Bark() ' From Dog class

        ' Multi-Level Inheritance
        Dim myBulldog As New Bulldog()
        myBulldog.Eat() ' From Animal class
        myBulldog.Bark() ' From Dog class
        myBulldog.ShowBreed() ' From Bulldog class

        Console.ReadLine()
    End Sub

End Module

File: Q6.cs
using System;

namespace InheritanceExample
{
    // Base class
    class Animal
    {
        public void Eat()
        {
            Console.WriteLine("The animal is eating.");
        }
    }

    // Single inheritance: Dog inherits from Animal
    class Dog : Animal
    {
        public void Bark()
        {
            Console.WriteLine("The dog is barking.");
        }
    }

    // Multi-level inheritance: Bulldog inherits from Dog
    class Bulldog : Dog
    {
        public void ShowBreed()
        {
            Console.WriteLine("This is a Bulldog.");
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            // Single Inheritance
            Dog myDog = new Dog();
            myDog.Eat(); // From Animal class
            myDog.Bark(); // From Dog class

            // Multi-Level Inheritance
            Bulldog myBulldog = new Bulldog();
            myBulldog.Eat(); // From Animal class
            myBulldog.Bark(); // From Dog class
            myBulldog.ShowBreed(); // From Bulldog class

            Console.ReadLine();
        }
    }
}
